### PR TITLE
Change prerelease versioning

### PR DESCRIPTION
**Pull request by weshaggard**
- Use alpha instead of dev
- Use beta instead of preview

Updating versioning logic to match guideline update in https://github.com/Azure/azure-sdk/pull/1536. 

PTAL @AlexGhiondea  @pakrym @heaths

### CODE REVIEW

File: doc/dev/Versioning.md [hunk 1]
```diff
@@ -9,25 +9,25 @@ Package version will look like:
...
-<Version>1.0.0-preview.1</Version>
+<Version>1.0.0-beta.1</Version>
 ```
 
-By default builds will replace any prerelease identifier with a `dev.yyyyMMdd.r` to ensure we have unique package versions. The date will come from either
+By default builds will replace any prerelease identifier with a `alpha.yyyyMMdd.r` to ensure we have unique package versions for dev builds. The date will come from either
 today's date or a property named `OfficialBuildId` which we will pass as part of our build pipelines.
 
-If we need to produce a package that is not a dev version and has the version in the project (i.e. stable or preview) then the `SkipDevBuildNumber` should
+If we need to produce a package that is not an dev package with an alpha version and has the version in the project (i.e. stable or beta) then the `SkipDevBuildNumber` should
```
Comment (heaths) [inline doc/dev/Versioning.md:21]:
```suggestion
If we need to produce a package that is not a dev package with an alpha version and has the version in the project (i.e. stable or beta) then the `SkipDevBuildNumber` should
```

**Review by pakrym — APPROVED** [review]
LGTM

**Review by heaths — APPROVED** [review]

File: eng/Versioning.targets [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <_BuildNumber>$(OfficialBuildId)</_BuildNumber>
     <_BuildNumber Condition="'$(OfficialBuildId)' == ''">$([System.DateTime]::Now.ToString(yyyyMMdd)).1</_BuildNumber>
     <HasReleaseVersion>true</HasReleaseVersion>
-    <HasReleaseVersion Condition="$(Version.Contains('preview'))">false</HasReleaseVersion>
+    <HasReleaseVersion Condition="$(Version.Contains('preview')) or $(Version.Contains('beta'))">false</HasReleaseVersion>
```
Comment (AlexGhiondea) [inline eng/Versioning.targets:7]: Should this check for alpha as well?

Comment (weshaggard) [reply eng/Versioning.targets:7]: No as that should never be committed into the project file.

**Pull request merged**